File: api/path_move_all_versions.go
package vaku

import (
	"errors"
)

var (
	// ErrPathMoveAllVersions when PathMoveAllVersions fails.
	ErrPathMoveAllVersions = errors.New("path move all versions")
)

// PathMoveAllVersions moves all versions of a secret from source to destination (copy + delete).
// Only works on v2 kv engines.
func (c *Client) PathMoveAllVersions(src, dst string) error {
	err := c.PathCopyAllVersions(src, dst)
	if err != nil {
		return newWrapErr("", ErrPathMoveAllVersions, err)
	}

	err = c.PathDeleteMeta(src)
	if err != nil {
		return newWrapErr(src, ErrPathMoveAllVersions, err)
	}

	return nil
}

File: src/api/path_move_all_versions.rs
//! `PathMoveAllVersions` — copy all versions then delete source metadata.

use crate::api::client::Client;
use crate::api::error::{Error, ErrorKind};

impl Client {
    /// Move every version of `src` to `dst`. KV v2 only.
    pub async fn path_move_all_versions(&self, src: &str, dst: &str) -> Result<(), Error> {
        self.path_copy_all_versions(src, dst)
            .await
            .map_err(|e| Error::wrap("", ErrorKind::PathMoveAllVersions, Some(Box::new(e))))?;
        self.path_delete_meta(src)
            .await
            .map_err(|e| Error::wrap(src, ErrorKind::PathMoveAllVersions, Some(Box::new(e))))
    }
}
